### PR TITLE
Use log.Component in comp/forwarder folder

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

Use `log.Component` in `comp/forwarder` folder (Remove package `"github.com/DataDog/datadog-agent/pkg/util/log"` from `comp/forwarder`)

### Describe how to test/QA your changes

Check [client trace](https://github.com/DataDog/datadog-agent/blob/ba07b6f91a6e658166a94ab6e2a28033e41988e6/comp/forwarder/defaultforwarder/transaction/transaction.go#L79-L121) code is called.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: comp/dogstatsd/server/server_test.go [hunk 1]
```diff
@@ -489,7 +493,8 @@ func TestHistToDist(t *testing.T) {
 
 	deps := fulfillDepsWithConfigOverride(t, cfg)
 
-	demux := aggregator.InitTestAgentDemultiplexerWithFlushInterval(10 * time.Millisecond)
+	log := fxutil.Test[log.Component](t, log.MockModule)
```
Comment (gh123man) [inline comp/dogstatsd/server/server_test.go:496]: the log dependency exists on the `serverDeps` object, so you can omit this line and just use `deps.Log` below. 
(This occurs a few times in this file)

**Review by modernplumbing — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]

**Review by truthbk — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]
/lgtm for @DataDog/container-integrations  files

**Review by xlucas — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]
Looks good for @DataDog/platform-integrations files.

**Review by maxday — APPROVED** [review]
LGTM for serverless-owned files

**Review by kkhor-datadog — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]

**Pull request merged**